infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -567,6 +567,59 @@ public void setLogEntrySyncOngoing(String clusterId, long remainingEntriesToSend
         }
     }
 
+    /**
+     * Update replication status table's prior sync info as COMPLETED.
+     *
+     * @param clusterId standby cluster id
+     * @param baseSnapshot Corfu log timestamp at which the last snapshot was based
+     */
+    public void setPriorSnapshotInfoComplete(String clusterId, long baseSnapshot) {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+        ReplicationStatusVal updatedStatusVal;
+
+        try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
+
+            CorfuStoreEntry<ReplicationStatusKey, ReplicationStatusVal, Message> record = txn.getRecord(replicationStatusTable, key);
+
+            if (record.getPayload() == null) {
+                log.error("Remote Cluster {} not found in Replication Status Table.", clusterId);
+                return;
+            }
+
+            ReplicationStatusVal previousStatusVal = record.getPayload();
+            SnapshotSyncInfo previousSyncInfo = previousStatusVal.getSnapshotSyncInfo();
+            SnapshotSyncInfo updatedSyncInfo;
+
+            if (previousSyncInfo.getStatus().equals(SyncStatus.COMPLETED)) {
+                log.info("Previous sync already marked complete, skipping update of previous sync info.");
+                return;
+            }
+
+            Instant time = Instant.now();
+            Timestamp timestamp = Timestamp.newBuilder().setSeconds(time.getEpochSecond())
+                    .setNanos(time.getNano()).build();
+            updatedSyncInfo = previousSyncInfo.toBuilder()
+                    .setStatus(SyncStatus.COMPLETED)
+                    .setBaseSnapshot(baseSnapshot)
+                    .setCompletedTime(timestamp)
+                    .build();
+
+            snapshotSyncTimerSample
+                    .flatMap(sample -> MeterRegistryProvider.getInstance()
+                            .map(registry -> {
+                                Timer timer = registry.timer("logreplication.snapshot.duration");
+                                return sample.stop(timer);
+                            }));
```

Comment:
Wondering if the timer is not started, what would happen on invoking stop?
like : say instead of sink changing leadership, what if source restarts/leader changes